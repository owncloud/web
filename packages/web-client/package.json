{
  "name": "@ownclouders/web-client",
  "version": "11.3.1",
  "description": "ownCloud web client",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-client",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-client"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "default": "./dist/web-client.js",
        "require": "./dist/web-client.cjs",
        "types": "./dist/src/index.d.ts"
      },
      "./graph": {
        "default": "./dist/web-client/graph.js",
        "require": "./dist/web-client/graph.cjs",
        "types": "./dist/src/graph/index.d.ts"
      },
      "./graph/generated": {
        "default": "./dist/web-client/graph/generated.js",
        "require": "./dist/web-client/graph/generated.cjs",
        "types": "./dist/src/graph/generated/index.d.ts"
      },
      "./ocs": {
        "default": "./dist/web-client/ocs.js",
        "require": "./dist/web-client/ocs.cjs",
        "types": "./dist/src/ocs/index.d.ts"
      },
      "./sse": {
        "default": "./dist/web-client/sse.js",
        "require": "./dist/web-client/sse.cjs",
        "types": "./dist/src/sse/index.d.ts"
      },
      "./webdav": {
        "default": "./dist/web-client/webdav.js",
        "require": "./dist/web-client/webdav.cjs",
        "types": "./dist/src/webdav/index.d.ts"
      }
    }
  },
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./graph": {
      "default": "./src/graph/index.ts"
    },
    "./graph/generated": {
      "default": "./src/graph/generated/index.ts"
    },
    "./ocs": {
      "default": "./src/ocs/index.ts"
    },
    "./sse": {
      "default": "./src/sse/index.ts"
    },
    "./webdav": {
      "default": "./src/webdav/index.ts"
    },
    "./l10n": {
      "default": "./l10n/translations.json"
    }
  },
  "scripts": {
    "generate-openapi": "rm -rf src/graph/generated && docker run --rm -v \"${PWD}/src/graph:/local\" openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/owncloud/libre-graph-api/main/api/openapi-spec/v1.0.yaml  -g typescript-axios -o /local/generated",
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/web-pkg": "workspace:*",
    "axios": "^1.7.7",
    "fast-xml-parser": "^4.5.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "uuid": "^11.0.0",
    "webdav": "^5.7.1",
    "xml-js": "^1.6.11",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@types/luxon": "3.4.2",
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0",
    "vite": "5.4.8"
  }
}
