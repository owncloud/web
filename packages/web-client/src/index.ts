import { graph } from './graph'
import { ocs } from './ocs'
import { webdav } from './webdav'

export * from './errors'
export * from './helpers'
export * from './utils'
export * from './constants'

export { graph, ocs, webdav }
