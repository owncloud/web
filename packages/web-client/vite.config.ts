import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'

const projectRootDir = searchForWorkspaceRoot(process.cwd())

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  build: {
    lib: {
      entry: {
        'web-client': resolve(__dirname, 'src/index.ts'),
        'web-client/graph': resolve(__dirname, 'src/graph/index.ts'),
        'web-client/graph/generated': resolve(__dirname, 'src/graph/generated/index.ts'),
        'web-client/ocs': resolve(__dirname, 'src/ocs/index.ts'),
        'web-client/sse': resolve(__dirname, 'src/sse/index.ts'),
        'web-client/webdav': resolve(__dirname, 'src/webdav/index.ts')
      }
    }
  },
  plugins: [
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({
      include: ['src'],
      outDir: 'dist/types',
      insertTypesEntry: true
    })
  ]
})
