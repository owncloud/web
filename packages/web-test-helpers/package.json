{
  "name": "@ownclouders/web-test-helpers",
  "version": "11.3.2",
  "description": "ownCloud web test helpers",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "main": "src/index.ts",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-test-helpers",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-test-helpers"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-test-helpers.es.js",
        "require": "./dist/web-test-helpers.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "peerDependencies": {
    "@ownclouders/web-pkg": "workspace:^",
    "@vue/test-utils": "^2.4.6",
    "vue": "^3.5.10"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "^5.0.0",
    "vite-plugin-dts": "^4.2.3",
    "vite": "^5.4.8"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@pinia/testing": "^0.1.3",
    "axios": "1.7.7",
    "pinia": "2.2.6",
    "vitest-mock-extended": "2.0.2",
    "vue-router": "4.2.5",
    "vue3-gettext": "2.4.0"
  }
}
