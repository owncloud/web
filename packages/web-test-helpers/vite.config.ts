import { resolve } from 'path'
import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import pkg from './package.json' assert { type: 'json' }
import vue from '@vitejs/plugin-vue'

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-test-helpers',
      fileName: (format) => `web-test-helpers.${format}.js`
    },
    rollupOptions: {
      external: [...Object.keys(pkg.dependencies), ...Object.keys(pkg.peerDependencies)]
    }
  },
  plugins: [vue(), dts({ include: ['src'], outDir: 'dist/types', insertTypesEntry: true })]
})
