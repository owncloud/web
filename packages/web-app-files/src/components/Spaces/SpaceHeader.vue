<template>
  <div
    class="space-header oc-p-m"
    :class="{ 'oc-flex': !imageExpanded && !isMobileWidth, 'space-header-squashed': isSideBarOpen }"
  >
    <div
      class="space-header-image"
      :class="{ 'space-header-image-expanded': imageExpanded || isMobileWidth }"
    >
      <img
        v-if="imageContent"
        class="oc-cursor-pointer"
        alt=""
        :src="imageContent"
        @click="toggleImageExpanded"
      />
      <div v-else class="space-header-image-default oc-flex oc-flex-middle oc-flex-center">
        <oc-icon name="layout-grid" size="xxlarge" class="oc-px-m oc-py-m" />
      </div>
    </div>
    <div class="space-header-infos">
      <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
        <div class="oc-flex oc-flex-middle space-header-infos-heading">
          <h2 class="space-header-name">{{ space.name }}</h2>
          <oc-button
            :id="`space-context-btn`"
            v-oc-tooltip="$gettext('Show context menu')"
            :aria-label="$gettext('Show context menu')"
            appearance="raw"
            class="oc-ml-s"
          >
            <oc-icon name="more-2" />
          </oc-button>
          <oc-drop
            :drop-id="`space-context-drop`"
            :toggle="`#space-context-btn`"
            mode="click"
            close-on-click
            :options="{ delayHide: 0 }"
            padding-size="small"
            position="right-start"
          >
            <space-context-actions :action-options="{ resources: [space] }" />
          </oc-drop>
        </div>
        <oc-button
          v-if="memberCount"
          :aria-label="$gettext('Open context menu and show members')"
          appearance="raw"
          @click="openSideBarSharePanel"
        >
          <oc-icon name="group" fill-type="line" size="small" />
          <span class="space-header-people-count oc-text-small" v-text="memberCountString"></span>
        </oc-button>
      </div>
      <p v-if="space.description" class="oc-mt-rm oc-text-bold">{{ space.description }}</p>
      <div
        v-if="markdownResource && markdownContent"
        ref="markdownContainerRef"
        class="markdown-container oc-flex oc-flex-middle"
      >
        <text-editor is-read-only :current-content="markdownContent" />
        <div class="markdown-container-edit oc-ml-s">
          <router-link
            v-oc-tooltip="$gettext('Edit description')"
            size="small"
            appearance="raw"
            :to="editReadMeContentLink"
          >
            <oc-icon name="pencil" size="small" fill-type="line" />
          </router-link>
        </div>
      </div>
      <div
        v-if="showMarkdownCollapse && markdownContent"
        class="markdown-collapse oc-text-center oc-mt-s"
      >
        <oc-button appearance="raw" @click="toggleMarkdownCollapsed">
          <oc-icon :name="toggleMarkdownCollapsedIcon" />
          <span>{{ toggleMarkdownCollapsedText }}</span>
        </oc-button>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import {
  computed,
  defineComponent,
  inject,
  onBeforeUnmount,
  onMounted,
  PropType,
  Ref,
  ref,
  unref,
  watch
} from 'vue'
import { buildSpaceImageResource, SpaceResource } from '@ownclouders/web-client'
import {
  useClientService,
  ProcessorType,
  useResourcesStore,
  useFileActions,
  useLoadPreview,
  TextEditor
} from '@ownclouders/web-pkg'
import { ImageDimension } from '@ownclouders/web-pkg'
import SpaceContextActions from './SpaceContextActions.vue'
import { eventBus } from '@ownclouders/web-pkg'
import { SideBarEventTopics } from '@ownclouders/web-pkg'
import { useGettext } from 'vue3-gettext'
import { DriveItem } from '@ownclouders/web-client/graph/generated'

const markdownContainerCollapsedClass = 'collapsed'

export default defineComponent({
  name: 'SpaceHeader',
  components: {
    SpaceContextActions,
    TextEditor
  },
  props: {
    space: {
      type: Object as PropType<SpaceResource>,
      required: true
    },
    isSideBarOpen: { type: Boolean, default: false }
  },
  setup(props) {
    const language = useGettext()
    const { $gettext, $ngettext } = language
    const clientService = useClientService()
    const { getFileContents, getFileInfo } = clientService.webdav
    const resourcesStore = useResourcesStore()
    const { getDefaultAction } = useFileActions()
    const { loadPreview } = useLoadPreview()

    const markdownContainerRef = ref(null)
    const markdownContent = ref('')
    const markdownResource = ref(null)
    const markdownCollapsed = ref(true)
    const showMarkdownCollapse = ref(false)
    const toggleMarkdownCollapsedIcon = computed(() => {
      return unref(markdownCollapsed) ? 'add' : 'subtract'
    })
    const toggleMarkdownCollapsedText = computed(() => {
      return unref(markdownCollapsed) ? $gettext('Show more') : $gettext('Show less')
    })
    const toggleMarkdownCollapsed = () => {
      markdownCollapsed.value = !unref(markdownCollapsed)
      unref(markdownContainerRef).classList.toggle(markdownContainerCollapsedClass)
    }
    const onMarkdownResize = () => {
      if (!unref(markdownContainerRef)) {
        return
      }

      unref(markdownContainerRef).classList.remove(markdownContainerCollapsedClass)
      const markdownContainerHeight = unref(markdownContainerRef).offsetHeight
      if (markdownContainerHeight < 150) {
        showMarkdownCollapse.value = false
        return
      }
      showMarkdownCollapse.value = true

      if (unref(markdownCollapsed)) {
        unref(markdownContainerRef).classList.add(markdownContainerCollapsedClass)
      }
    }
    const markdownResizeObserver = new ResizeObserver(onMarkdownResize)
    const observeMarkdownContainerResize = () => {
      if (!markdownResizeObserver || !unref(markdownContainerRef)) {
        return
      }
      markdownResizeObserver.unobserve(unref(markdownContainerRef))
      markdownResizeObserver.observe(unref(markdownContainerRef))
    }
    const unobserveMarkdownContainerResize = () => {
      if (!markdownResizeObserver || !unref(markdownContainerRef)) {
        return
      }
      markdownResizeObserver.unobserve(unref(markdownContainerRef))
    }
    onMounted(observeMarkdownContainerResize)
    onBeforeUnmount(() => {
      unobserveMarkdownContainerResize()
    })
    watch(
      computed(() => props.space.spaceReadmeData),
      async (data: DriveItem) => {
        if (!data) {
          return
        }

        const fileContentsResponse = await getFileContents(props.space, {
          path: `.space/${props.space.spaceReadmeData.name}`
        })

        const fileInfoResponse = await getFileInfo(props.space, {
          path: `.space/${props.space.spaceReadmeData.name}`
        })

        unobserveMarkdownContainerResize()
        markdownContent.value = fileContentsResponse.body
        markdownResource.value = fileInfoResponse

        if (unref(markdownContent)) {
          observeMarkdownContainerResize()
        }
      },
      { deep: true, immediate: true }
    )

    const imageContent = ref<string>(null)
    const imageExpanded = ref(false)

    const editReadMeContentLink = computed(() => {
      const action = getDefaultAction({ resources: [unref(markdownResource)], space: props.space })

      if (!action.route) {
        return null
      }

      return action.route({ space: props.space, resources: [unref(markdownResource)] })
    })
    const toggleImageExpanded = () => {
      imageExpanded.value = !unref(imageExpanded)
    }

    watch(
      computed(() => props.space.spaceImageData),
      async (data) => {
        if (!data) {
          return
        }
        const resource = buildSpaceImageResource(props.space)
        imageContent.value = await loadPreview({
          space: props.space,
          resource,
          dimensions: ImageDimension.Tile,
          processor: ProcessorType.enum.fit,
          cancelRunning: true,
          updateStore: false
        })
      },
      { immediate: true }
    )

    const memberCount = computed(() => {
      return Object.keys(props.space.members).length
    })
    const memberCountString = computed(() => {
      return $ngettext('%{count} member', '%{count} members', unref(memberCount), {
        count: unref(memberCount).toString()
      })
    })

    const openSideBarSharePanel = () => {
      resourcesStore.setSelection([])
      eventBus.publish(SideBarEventTopics.openWithPanel, 'space-share')
    }

    return {
      isMobileWidth: inject<Ref<boolean>>('isMobileWidth'),
      markdownContainerRef,
      markdownContent,
      markdownResource,
      markdownCollapsed,
      showMarkdownCollapse,
      toggleMarkdownCollapsedIcon,
      toggleMarkdownCollapsedText,
      toggleMarkdownCollapsed,
      imageContent,
      imageExpanded,
      toggleImageExpanded,
      memberCount,
      memberCountString,
      openSideBarSharePanel,
      editReadMeContentLink
    }
  }
})
</script>

<style lang="scss">
.space-header {
  &-squashed {
    .space-header-image {
      @media only screen and (max-width: 1200px) {
        display: none;
      }
    }
  }

  &-image {
    width: 280px;
    min-width: 280px;
    aspect-ratio: 16 / 9;
    margin-right: var(--oc-space-large);
    max-height: 158px;

    &-default {
      background-color: var(--oc-color-background-highlight);
      height: 100%;
      border-radius: 10px;
    }

    &-expanded {
      width: 100%;
      margin: 0;
      max-height: 100%;
      max-width: 100%;
    }

    img {
      border-radius: 10px;
      height: 100%;
      width: 100%;
      max-height: 100%;
      object-fit: cover;
    }
  }

  &-infos {
    flex: 1;

    &-heading {
      max-width: 100%;
    }
  }

  &-name {
    font-size: 1.5rem;
    word-break: break-all;
  }

  &-people-count {
    white-space: nowrap;
  }

  .markdown-container.collapsed {
    max-height: 100px;
    overflow: hidden;
    -webkit-mask-image: linear-gradient(180deg, #000 90%, transparent);
  }
}
</style>
