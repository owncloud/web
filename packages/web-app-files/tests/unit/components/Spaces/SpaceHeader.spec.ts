import { nextTick, ref } from 'vue'
import SpaceHeader from 'web-app-files/src/components/Spaces/SpaceHeader.vue'
import { Drive } from '@ownclouders/web-client/graph/generated'
import { SpaceResource, buildSpace, Resource } from '@ownclouders/web-client'
import { defaultPlugins, mount, defaultComponentMocks } from 'web-test-helpers'
import { mock } from 'vitest-mock-extended'

vi.mock('@ownclouders/web-pkg', async (importOriginal) => ({
  ...(await importOriginal<any>()),
  useFileActions: vi.fn().mockReturnValue({
    getDefaultAction: vi.fn().mockReturnValue({ handler: vi.fn() })
  })
}))

describe('SpaceHeader', () => {
  it('should add the "squashed"-class when the sidebar is opened', () => {
    const wrapper = getWrapper({
      space: buildSpace({ id: '1' } as unknown as Drive),
      isSideBarOpen: true
    })
    expect(wrapper.find('.space-header-squashed').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  describe('space image', () => {
    it('should show the default image if no other image is set', () => {
      const wrapper = getWrapper({ space: buildSpace({ id: '1' } as unknown as Drive) })
      expect(wrapper.find('.space-header-image-default').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('should show the set image', async () => {
      const spaceMock = { spaceImageData: { webDavUrl: '/' } }
      const wrapper = getWrapper({
        space: { ...buildSpace({ id: '1' } as unknown as Drive), ...spaceMock }
      })
      await wrapper.vm.loadPreviewTask.last
      expect(wrapper.find('.space-header-image-default').exists()).toBeFalsy()
      expect(wrapper.find('.space-header-image img').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('should take full width in mobile view', () => {
      const spaceMock = { spaceImageData: { webDavUrl: '/' } }
      const wrapper = getWrapper({
        space: { ...buildSpace({ id: '1' } as unknown as Drive), ...spaceMock },
        isMobileWidth: true
      })
      expect(wrapper.find('.space-header').classes()).not.toContain('oc-flex')
      expect(wrapper.find('.space-header-image').classes()).toContain('space-header-image-expanded')
    })
  })
  describe('space description', () => {
    it('should show the description', async () => {
      const wrapper = getWrapper({ space: buildSpace({ id: '1' } as unknown as Drive) })
      wrapper.vm.markdownResource = mock<Resource>()
      wrapper.vm.markdownContent = 'content'
      await nextTick()
      expect(wrapper.find('.markdown-container').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})

function getWrapper({ space = {} as SpaceResource, isSideBarOpen = false, isMobileWidth = false }) {
  const mocks = defaultComponentMocks()
  mocks.$previewService.loadPreview.mockResolvedValue('blob:image')
  return mount(SpaceHeader, {
    props: {
      space,
      isSideBarOpen
    },
    global: {
      mocks,
      plugins: [...defaultPlugins()],
      provide: { ...mocks, isMobileWidth: ref(isMobileWidth) },
      stubs: {
        'quota-modal': true,
        'space-context-actions': true
      }
    }
  })
}
