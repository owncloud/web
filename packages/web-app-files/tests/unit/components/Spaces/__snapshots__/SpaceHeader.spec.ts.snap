// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpaceHeader > should add the "squashed"-class when the sidebar is opened 1`] = `
"<div class="space-header oc-p-m oc-flex space-header-squashed">
  <div class="space-header-image">
    <div class="space-header-image-default oc-flex oc-flex-middle oc-flex-center"><span class="oc-icon oc-icon-xxl oc-icon-passive oc-px-m oc-py-m"><!----></span></div>
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name"></h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m oc-icon-passive"><!----></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]"></space-context-actions-stub>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`SpaceHeader > space description > should show the description 1`] = `
"<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image">
    <div class="space-header-image-default oc-flex oc-flex-middle oc-flex-center"><span class="oc-icon oc-icon-xxl oc-icon-passive oc-px-m oc-py-m"><!----></span></div>
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name"></h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m oc-icon-passive"><!----></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]"></space-context-actions-stub>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <div class="markdown-container oc-flex oc-flex-middle">
      <!---->
      <div class="markdown-container-edit oc-ml-s"><a attrs="[object Object]" size="small" appearance="raw"></a></div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`SpaceHeader > space image > should show the default image if no other image is set 1`] = `
"<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image">
    <div class="space-header-image-default oc-flex oc-flex-middle oc-flex-center"><span class="oc-icon oc-icon-xxl oc-icon-passive oc-px-m oc-py-m"><!----></span></div>
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name"></h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m oc-icon-passive"><!----></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]"></space-context-actions-stub>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`SpaceHeader > space image > should show the set image 1`] = `
"<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image"><img class="oc-cursor-pointer" alt="" src="blob:image"></div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h2 class="space-header-name"></h2> <button type="button" aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn">
          <!--v-if-->
          <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m oc-icon-passive"><!----></span>
        </button>
        <div id="space-context-drop" class="oc-drop oc-box-shadow-medium oc-rounded" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <space-context-actions-stub actionoptions="[object Object]"></space-context-actions-stub>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
